src/Discord/Internal/Types/ScheduledEvents.hs

```diff
@@ -0,0 +1,586 @@
+      then Nothing
+      else
+        let Just loc' = modifyScheduledEventDataLocation
+        in  if isNothing loc'
+              then Just Null
+              else
+                let Just loc'' = loc'
+                in  Just $ object [("location", toJSON loc'')]
+    desc = if isNothing modifyScheduledEventDataDescription
+      then Nothing
+      else
+        let Just desc' = modifyScheduledEventDataDescription
+        in  if isNothing desc'
+              then Just Null
+              else let Just desc'' = desc' in Just $ toJSON desc''
+
+instance FromJSON ModifyScheduledEventData where
+  parseJSON = withObject
+    "ModifyScheduledEventData"
+    (\v -> do
+      -- The trivial fields
+      msename  <- v .:? "name"
+      msest    <- v .:? "scheduled_start_time"
+      mseet    <- v .:? "scheduled_end_time"
+      msetype  <- v .:? "entity_type"
+      msepl    <- v .:? "privacy_level"
+      msestat  <- v .:? "status"
+      mseimg   <- v .:? "image"
+
+      -- The not so trivial ones
+      msecid'  <- v .:! "channel_id"
+      mseloc'  <- v .:! "entity_metadata"
+      msedesc' <- v .:! "description"
+
+      -- Extract the values
+      msecid   <- case msecid' of
+        Nothing   -> return Nothing
+        Just Null -> return $ Just Nothing
+        Just x    -> do
+          x' <- parseJSON x
+          return $ Just x'
+
+      mseloc <- case mseloc' of
+        Nothing   -> return Nothing
+        Just Null -> return $ Just Nothing
+        Just x    -> do
+          x' <- withObject "entity_metadata" (.: "location") x
+          return $ Just x'
+
+      msedesc <- case msedesc' of
+        Nothing   -> return Nothing
+        Just Null -> return $ Just Nothing
+        Just x    -> do
+          x' <- parseJSON x
+          return $ Just x'
+
+      return $ ModifyScheduledEventData
+        { modifyScheduledEventDataChannelId    = msecid
+        , modifyScheduledEventDataLocation     = mseloc
+        , modifyScheduledEventDataName         = msename
+        , modifyScheduledEventDataPrivacyLevel = msepl
+        , modifyScheduledEventDataStartTime    = msest
+        , modifyScheduledEventDataEndTime      = mseet
+        , modifyScheduledEventDataDescription  = msedesc
+        , modifyScheduledEventDataType         = msetype
+        , modifyScheduledEventDataStatus       = msestat
+        , modifyScheduledEventDataImage        = mseimg
+        }
+    )
+
+-- | An User that subscribed to a Scheduled Event
+data ScheduledEventUser = ScheduledEventUser
+  { scheduledEventUserEvent       :: ScheduledEventId
+  , scheduledEventUserUser        :: User
+  , scheduledEventUserGuildMember :: Maybe GuildMember
+  }
+
+instance FromJSON ScheduledEventUser where
+  parseJSON = withObject
+    "ScheduledEventUser"
+    (\v -> do
+      seue <- v .: "guild_scheduled_event_id"
+      seuu <- v .: "user"
+      seum <- v .:? "member"
+      return $ ScheduledEventUser seue seuu seum
+    )
```

Comment:
Minor thing, this could use the pattern we use elsewhere with the constructor at the top followed by <$> and <*>'s.

`ScheduledEventUser
    <$> v .: "guild_scheduled_event_id"
    <*> v .: "user"
    <*> v .:? "member"`